saleor/graphql/order/bulk_mutations/order_bulk_create.py

```diff
@@ -0,0 +1,1067 @@
+            total_price_gross_amount=line_amounts.total_gross,
+            undiscounted_unit_price_net_amount=line_amounts.undiscounted_unit_net,
+            undiscounted_unit_price_gross_amount=line_amounts.undiscounted_unit_gross,
+            undiscounted_total_price_net_amount=line_amounts.undiscounted_total_net,
+            undiscounted_total_price_gross_amount=line_amounts.undiscounted_total_gross,
+            tax_rate=line_amounts.tax_rate,
+            tax_class=line_tax_class,
+            tax_class_name=tax_class_name,
+        )
+
+        if metadata := order_line_input.get("tax_class_metadata"):
+            for data in metadata:
+                order_line.tax_class_metadata.update({data["key"]: data["value"]})
+        if private_metadata := order_line_input.get("tax_class_private_metadata"):
+            for data in private_metadata:
+                order_line.tax_class_private_metadata.update(
+                    {data["key"]: data["value"]}
+                )
+        return order_line
+
+    @classmethod
+    def create_single_order(
+        cls, order_input, object_storage: Dict[str, Any]
+    ) -> OrderWithErrors:
+        order = OrderWithErrors(
+            order=None,
+            order_errors=[],
+            lines=[],
+            lines_errors=[],
+            notes=[],
+            notes_errors=[],
+        )
+        cls.validate_order_input(order_input, order.order_errors)
+        order_instance = Order()
+
+        # get order related instances
+        instances = cls.get_instances_related_to_order(
+            order_input=order_input,
+            errors=order.order_errors,
+            object_storage=object_storage,
+        )
+        delivery_input = order_input["delivery_method"]
+        delivery_method = cls.get_delivery_method(
+            input=delivery_input,
+            errors=order.order_errors,
+            object_storage=object_storage,
+        )
+        if not instances or not delivery_method:
+            return order
+
+        # create lines
+        order_lines_input = order_input["lines"]
+        for order_line_input in order_lines_input:
+            if order_line := cls.create_single_order_line(
+                order_line_input,
+                order_instance,
+                object_storage,
+                order_input,
+                order.lines_errors,
+            ):
+                order.lines.append(order_line)
+            else:
+                order.is_each_line_created = False
+
+        if not order.is_each_line_created:
+            order.order_errors.append(
+                OrderBulkError(
+                    message="At least one order line can't be created.",
+                    field="lines",
+                    code=OrderBulkCreateErrorCode.ORDER_LINE_ERROR,
+                )
+            )
+            return order
+        # TODO check if multiple order lines contains the same variant (fulfillments)
```

Comment:
What action should be taken regarding this TODO?

Reply:
This will be handled in separate PRs - description updated.